types/knex.d.ts

```diff
@@ -239,7 +239,7 @@ declare namespace Knex {
     (tableName: TableName | Identifier | QueryCallback, clause: JoinCallback): QueryBuilder;
     (
       tableName: TableName | Identifier | QueryCallback,
-      columns: { [key: string]: string | number | Raw }
+      columns: { [key: string]: string | number | boolean | Raw }
```

Comment:
When would it be a boolean?

Reply:
What do you mean? 
The value of join can be boolean :)

Reply:
Could you please provide example code that was invalid before this change?
The way I'm reading this binding is this:
```
knex('users')
  .join('contacts', 'users.id', '=', 'contacts.user_id')
  .select('users.id', 'contacts.phone')
```
`columns` are these: ` 'users.id', '=', 'contacts.user_id'`
Why would they ever be boolean?

Reply:
Pay attention to the interface that I've fixed, it is the one that accepts an object.

```javascript
db('tableName')
    .leftJoin('otherTable', {
      'tableName.isDefault': true
    })
``` 
![image](https://user-images.githubusercontent.com/9304194/55279572-f83d2780-532a-11e9-94f1-cf29c58788e6.png)

It tries to resolve it using the `Raw` interface because there is no boolean.

Reply:
Fair enough. And if typing is fixed, DB understands boolean value correctly?

Reply:
Yeah, it works, it converts it into tinyInt automatically

Reply:
Actually that depends highly on db that is used, but something like that :) I had big problems trying to write cross db compatible code that had used boolean column type.